src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala

```diff
@@ -212,13 +212,15 @@ final case class PurposeVersionActivation(
   )(implicit contexts: Seq[(String, String)]): Future[PurposeVersion] = {
     val documentId: UUID = uuidSupplier.get()
     for {
-      (producerDescription, consumerDescription) <- getTenantName(eService.producerId).zip(
-        getTenantName(purpose.consumerId)
-      )
+      ((producerDescription, producerValue), (consumerDescription, consumerValue)) <- getTenantTuple(
+        eService.producerId
+      ).zip(getTenantTuple(purpose.consumerId))
```

Comment:
IMO this became less readable, I would suggest something like
```suggestion
      (producer, consumer) <- getTenant(eService.producerId).zip(getTenant(purpose.consumerId))
```